src/Datadog.Trace.ClrProfiler.Native/clr_helpers.h

```diff
@@ -61,6 +61,8 @@ class EnumeratorIterator {
       if (status_ == S_OK && sz_ == 0) {
         status_ = S_FALSE;
       }
+    } else {
+      status_ = status;
```

Comment:
Here it is (as far as I can tell from testing). `status_` was not always initialized, which caused `FindAssemblyRef()` to not find assembly references... sometimes.